src/archive/mod.rs

```diff
@@ -224,80 +339,71 @@ impl<'a> Archive<'a> {
             return Err(Error::BadMagic(magic.pread(0)?).into());
         }
         let mut member_array = Vec::new();
-        let size = buffer.as_ref().len() as u64;
-        let mut pos = 0u64;
         let mut index = Index::default();
-        let mut extended_names = NameIndex::default();
-        loop {
-            if pos >= size { break }
-            // if the member is on an uneven byte boundary, we bump the buffer
-            if pos & 1 == 1 {
+        let mut sysv_name_index = NameIndex::default();
+        while *offset < buffer.len() {
+            // realign the cursor to a word boundary, if it's not on one already
```

Comment:
is eveness a word boundary ? Also I dunno what a word is :laughing: it's size is always shifting on me. No need to respond to this ;)